test/unit/controllers/VersionsControllerSpec.scala

```diff
@@ -0,0 +1,294 @@
+    when(mockDateTimeService.nowUtc()).thenReturn(dateTime)
+    when(mockApiSubscriptionFieldsConnector.getSubscriptionFields(any[ApiSubscriptionKey])(any[ValidatedHeadersRequest[_]], any[HeaderCarrier])).thenReturn(Future.successful(apiSubscriptionFieldsResponse))
+  }
+
+  private val errorResultBadgeIdentifier = errorBadRequest("X-Badge-Identifier header is missing or invalid").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+  private val errorResultMissingIdentifiers = errorBadRequest("Both X-Submitter-Identifier and X-Badge-Identifier headers are missing").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+
+  private val errorResultMrnTooLong = errorBadRequest("MRN too long").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+  private val errorResultMissingMrn = errorBadRequest("Missing MRN").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+  private val errorResultCDS60002 = errorBadRequest("MRN invalid", "CDS60002").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+
+  "Declaration Version Controller for MRN queries" should {
+    "process CSP request when call is authorised for CSP" in new SetUp() {
+      authoriseCsp()
+
+      awaitSubmitMrn(ValidCspDeclarationStatusRequest)
+
+      verifyCspAuthorisationCalled(numberOfTimes = 1)
+    }
+
+    "process CSP request when call is authorised for CSP and declarationSubmissionChannel is set" in new SetUp() {
+      authoriseCsp()
+
+      val result: Future[Result] = submitMrn(ValidCspDeclarationVersionRequestWithDeclarationSubmissionChannel, Some("AuthenticatedPartyOnly"))
+      status(result) shouldBe OK
+      verifyCspAuthorisationCalled(numberOfTimes = 1)
+    }
+
+    "process CSP request when call is authorised for CSP and declarationSubmissionChannel is invalid" in new SetUp() {
+      authoriseCsp()
+
+      val result: Future[Result] = submitMrn(ValidCspDeclarationVersionRequestWithInvalidDeclarationSubmissionChannel, Some("AuthenticatedPartyOnly1"))
+      status(result) shouldBe BAD_REQUEST
+      stringToXml(contentAsString(result)) shouldBe stringToXml("<errorResponse><code>BAD_REQUEST</code><message>Invalid declarationSubmissionChannel parameter</message></errorResponse>")
+    }
+
+    "respond with status 200 and conversationId in header for a processed valid CSP request" in new SetUp() {
+      authoriseCsp()
+
+      val result: Future[Result] = submitMrn(ValidCspDeclarationStatusRequest)
+
+      status(result) shouldBe OK
+      header(X_CONVERSATION_ID_NAME, result) shouldBe Some(conversationIdValue)
+    }
+
+    "respond with status 400 and conversationId in header for a request without an Mrn" in new SetUp() {
+      authoriseCsp()
+
+      val result: Future[Result] = controller.list("").apply(ValidCspDeclarationStatusRequest)
+
+      status(result) shouldBe BAD_REQUEST
+      stringToXml(contentAsString(result)) shouldBe stringToXml("<errorResponse><code>BAD_REQUEST</code><message>Missing MRN</message></errorResponse>")
```

Comment:
Message is incorrect. The word parameter is there usually. Others need fixing too.